build.ps1

```diff
@@ -868,11 +870,21 @@ $items.GetEnumerator() | %{
 	$item.Name = $name
 	$item.ArchiveFile = $archiveFile
 	$item.PatchDirectory = $(New-Object System.IO.DirectoryInfo $patchDirectory)
-	$item.BuildDirectory = $(New-Object System.IO.DirectoryInfo "$workingDirectory\$($archiveFile.BaseName)")
-	$item.BuildArchiveFile = $(New-Object System.IO.FileInfo "$workingDirectory\$($archiveFile.BaseName)-$filenameArch$($archiveFile.Extension)")
 	$item.Dependencies = @($item.Dependencies | %{ $items[$_] })
 	$item.Dependents = @()
 	$item.State = ''
+
+	switch ([System.IO.Path]::GetExtension($item.ArchiveFile)) {
+		'.xz' {
+			$filename = [System.IO.Path]::GetFileNameWithoutExtension($item.ArchiveFile)
```

Comment:
$item.ArchiveFile is $archiveFile, which is already a FileInfo. You don't need to convert it implicitly to a string and then use Path functions on it, just use $archiveFile.BaseName for this.